src/ast/mod.rs

```diff
@@ -4220,12 +4279,39 @@ pub struct HiveFormat {
     pub location: Option<String>,
 }
 
+/// Represents BigQuery specific configuration like partitioning, clustering
+/// information during table creation.
+///
+/// <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_table_statement>
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub struct BigQueryCreateTableConfiguration {
+    /// A partition expression for the table.
+    /// <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#partition_expression>
+    pub partition_by: Option<Expr>,
+    /// Table clustering column list.
+    /// <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#table_option_list>
+    pub cluster_by: Option<Vec<Ident>>,
+    /// Table options list.
+    /// <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#table_option_list>
+    pub options: Option<Vec<SqlOption>>,
+}
+
+/// Represents database specific configuration during table creation.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum CreateTableConfiguration {
```

Comment:
I wonder if you considered inlining these into `Statement::CreateTable`  rather than adding a new structure?

Like

```
/// CREATE TABLE
CreateTable {
    or_replace: bool,
    temporary: bool,
    external: bool,
    /// A partition expression for the table.
    /// <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#partition_expression>
    partition_by: Option<Expr>,
    /// Table clustering column list.
    /// <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#table_option_list>
     cluster_by: Option<Vec<Ident>>,
    /// Table options list.
    /// <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#table_option_list>
     options: Option<Vec<SqlOption>>,
...
 ```

I think that might be more consistent with other  dialect specific fields as well as `order_by` and `strict`.

Having an extra struct here also may be confusing as it doesn't seem to mirror the structure of the SQL -- for example,  it seems to imply to me that there is some sort of "CONFIG" keyword / clause that appears in the SQL when https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_table_statement seems to imply that `PARTITION BY` `CLUSTER BY` and `OPTIONS` are clauses of the `CREATE TABLE` statement itself

Reply:
Ah yes I figured to have them inline initially, agree that makes it less confusing. The only issue was with serializing back to SQL text - since the fields in bigquery's case have a defined order (e.g `PARTITION BY` before `CLUSTER BY`), the inlined fields might end up not being reusable by other dialects if they show up at different parts of the CREATE statement. I'm not so sure if that's worth optimizing for though, so happy to just inline them if that's preferrable

Reply:
I think an inlined version would be better to maintain consistency. If someone needs a different order when serializing for another dialect we can add that as a follow on PR perhaps.


If for some reason we need to keep the separate struct I think that would be acceptable if there were some doc comments explaining the rational (as that is not easy to understand from the code).

Reply:
That sounds good! Will make updates to this

Reply:
Inlined!